virtcontainers/fc.go

```diff
@@ -546,6 +555,77 @@ func (fc *firecracker) fcSetVMBaseConfig(mem int64, vcpus int64, htEnabled bool)
 	return err
 }
 
+func (fc *firecracker) fcSetLogger() error {
+	span, _ := fc.trace("fcSetLogger")
+	defer span.Finish()
+
+	fcLogLevel := "Error"
+
+	var err error
+	fcLogFifoFile := filepath.Join(fc.vmPath, fcLogFifo)
+	fc.logFifo, err = fifo.OpenFifo(context.Background(), fcLogFifoFile, syscall.O_CREAT|syscall.O_RDONLY|syscall.O_NONBLOCK, 0)
+	if err != nil {
+		return fmt.Errorf("Failed to open/create firecracker log fifo file")
+	}
+
+	jailedLogFifo, err := fc.fcJailResource(fcLogFifoFile, fcLogFifo)
+	if err != nil {
+		fc.Logger().WithField("set logger failed", err).Error()
+		return err
+	}
+
+	// listen on log fifo file and transfer error info
+	go func() {
+		scanner := bufio.NewScanner(fc.logFifo)
+		for scanner.Scan() {
+			fc.Logger().WithFields(logrus.Fields{
+				"level":    "log",
+				"contents": scanner.Text()}).Error("firecracker failed")
```

Comment:
@Pennyzct Why are you logging these as errors? Are these generic log messages or errors?

Reply:
Hi @amshinde Since it's the error info extracted from firecracker built-in logging scheme. It will only have outputs when firecracker failed setting up vmm. I have already set the logging level as `Error` here.
```
fcLogLevel := "Error"
```